Implementing_Video_Conferencing_with_ReactJS,_Node.md

```diff
@@ -0,0 +1,599 @@
+- Microphones and Webcams
+- Display screen, monitor or projector
+- Software-based coding and decoding technologies (CODEC)
+- Acoustic echo cancellation (AEC) software for audio optimization and real-time communication.
+
+**Requirements for Web Packets Handling**
+For a network communication to succeed it is necessary to have a unified standard for defining the architecture of communication systems. A digital environment supporting multiple data types including audio and video significantly increases the efficiency of a video conferencing application including greater bandwidth utilization.
+
+**Implementation and Coding**
+The application being a full stack project is divided into two segments, namely
+
+1. The Client-Side and
+2. The Server-Side
+
+**Client-Side Setup - (index.js)**
+The Client interface is setup using ReactJS which is a lightweight frontend JavaScript library. the various pages and subdivision of the client interface includes the following
+
+**Step 1. Getting started with a new React app**
+npx
+
+    npx create-react-app react-video-conferencing-app
+
+npm
+
+    npm init react-app react-video-conferencing-app
+
+yarn
+
+    yarn create react-app react-video-conferencing-app
+
+the above command should get you started with a new react app with all the default dependencies appropriately installed.
+
+    cd react-video-conferencing-app
+    npm start
+
+the above command will change the directory to your new react app and get the development server running.
+
+**Step 2. Installing the required Client dependencies**
+For a successful development of the application a few dependencies must be installed to enable react process and perform specific instructions and they include the following.
+
+    npm install
+    Or
+    yarn add
+    "apollo-cache-inmemory": "^1.1.9",
+    "apollo-client": "^2.2.5",
+    "apollo-client-preset": "^1.0.8",
+    "apollo-link-http": "^1.4.0",
+    "apollo-link-schema": "^1.0.6",
+    "apollo-link-ws": "^1.0.7",
+    "apollo-utilities": "^1.0.10",
+    "classnames": "^2.2.5",
+    "react-apollo": "^2.0.4",
+    "react-dom": "^16.2.0",
+    "react-redux": "^5.0.7",
+    "react-router": "^4.2.0",
+    "react-router-config": "^1.0.0-beta.4",
+    "react-router-dom": "^4.2.2",
+    "react-stay-scrolled": "^2.1.1",
+    "redux": "^3.7.2",
+    "redux-actions": "^2.2.1",
+    "redux-devtools-extension": "^2.13.2",
+    "redux-thunk": "^2.2.0",
+    "socket.io": "^2.4.0",
+    "socket.io-client": "^2.0.4",
+    "socket.io-redis": "^5.2.0",
+    "socketio-jwt": "^4.5.0",
+    "style-loader": "^0.20.2",
+
+**Step 3. Setting up the Client Index file**
+This file is the main file for integrating the client and the server codes together. It also enables the initialization of React DOM element, the Apollo-Client elements, WebRTC adapter and the other facilities for the implementation of the application.  
+
+    import 'webrtc-adapter';
```

Comment:
```javascript
    import 'webrtc-adapter';
    import { InMemoryCache } from 'apollo-cache-inmemory';
    import { ApolloClient } from 'apollo-client';
    import { split } from 'apollo-client-preset';
    import { HttpLink } from 'apollo-link-http';
    import { WebSocketLink } from 'apollo-link-ws';
    import { getMainDefinition } from 'apollo-utilities';
    import React from 'react';
    import { ApolloProvider } from 'react-apollo';
    import ReactDOM from 'react-dom';
    import { Provider } from 'react-redux';
    import { renderRoutes } from 'react-router-config';
    import { BrowserRouter } from 'react-router-dom';
    import './styles/index.scss';
    import routes from './routes';
    import store from './store';
    import { setToken } from './actions/token';
    store.dispatch(setToken(window.__JWT_TOKEN__));

    const httpLink = new HttpLink({
      uri: process.env.GRAPHQL_URI,
      credentials: 'same-origin',
    });
    const wsLink = new WebSocketLink({
      uri: process.env.GRAPHQL_WS_URI,
      options: {
        reconnect: true,
      },
    });
    const subscriptionMiddleware = {
      applyMiddleware(options, next) {
        const { token } = store.getState();
        options.connectionParams = { authToken: token };
        next();
      },
    };
    wsLink.subscriptionClient.use([subscriptionMiddleware]);
    const link = split(
      ({ query }) => {
        const { kind, operation } = getMainDefinition(query);
        return kind === 'OperationDefinition' && operation === 'subscription';
      },
      wsLink,
      httpLink,
    );
    const cache = new InMemoryCache().restore(window.__APOLLO_STATE__);
    const client = new ApolloClient({ link, cache, connectToDevTools: process.env.NODE_ENV === 'development' });
    delete window.__APOLLO_STATE__;
    delete window.__JWT_TOKEN__;

      render() {
        return (
          <Provider store={store}>
            <ApolloProvider client={client}>
              <BrowserRouter>
                {renderRoutes(routes, { userAgent: navigator.userAgent })}
              </BrowserRouter>
            </ApolloProvider>
          </Provider>
        );
      }
    }
    function render() {
      ReactDOM.hydrate(
        <Routes />,
        document.getElementById('entry-point')
    }
    render();
```
Do this for all the dependecies and programs. After the backticks, you should identify what the language is and close with the backticks.

Reply:
Please, I don't quite understand what is expected over here

Reply:
Mention the programming language after the opening backticks.  Something like this *```Javascript*.

Reply:
Done